cloud/scope/common_test.go

```diff
@@ -0,0 +1,188 @@
+package scope
+
+import (
+	"context"
+	"errors"
+	"testing"
+
+	mock "github.com/linode/cluster-api-provider-linode/mock"
+	"github.com/linode/linodego"
+	"github.com/stretchr/testify/assert"
+	"go.uber.org/mock/gomock"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// Test_createLinodeClient tests the createLinodeClient function. Checks if the client does not error out.
+func Test_createLinodeClient(t *testing.T) {
+	t.Parallel()
+
+	tests := []struct {
+		name   string
+		apiKey string
+		want   *linodego.Client
+	}{
+		{
+			"Valid API Key",
+			"test-key",
+			createLinodeClient("test-key"),
+		},
+		{
+			"Empty API Key",
+			"",
+			createLinodeClient(""),
+		},
+	}
+
+	for _, tt := range tests {
+		testCase := tt
+		t.Run(testCase.name, func(t *testing.T) {
+			t.Parallel()
+
+			if got := createLinodeClient(testCase.apiKey); got != nil {
+				assert.EqualExportedValues(t, testCase.want, got, "Checking is the objects are equal")
+			}
+		})
+	}
+}
+
+// Test_getCredentialDataFromRef tests the getCredentialDataFromRef function.
+func Test_getCredentialDataFromRef(t *testing.T) {
+	t.Parallel()
+
+	type args struct {
+		providedCredentialsRef corev1.SecretReference
+		expectedCredentialsRef corev1.SecretReference
+		defaultNamespace       string
```

Comment:
It doesn't seem like we need to parameterize `defaultNamespace` here since it's being assigned to `"default"` in all test cases.